sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py

```diff
@@ -259,10 +264,18 @@ def _get_code_asset_arm_id(self, code_asset: Code, register_asset: bool = True)
             if register_asset:
                 code_asset = self._code_assets.create_or_update(code_asset)
                 return code_asset.id
-            uploaded_code_asset, _ = _check_and_upload_path(
+            path, ignore_file, _ = _get_snapshot_path_info(code_asset)
+            workspace_info = self._datastore_operation._service_client.workspaces.get(
```

Comment:
You should have access to workspace operations in `_operation_container` here.

Reply:
Do you mean to call WorkspaceOperations.get()? If so, that's not the same call as what I'm doing here. 

The API call needs the workspace's ID as defined in the properties section of the UI. Our WorkspaceOperations.get() call does not return that. For ID, it returns this '/subscriptions/<sub id>/resourceGroups/<rg name/providers/Microsoft.MachineLearningServices/workspaces/<ws name>', so I had to find out a way to access the actual ID. Getting the workspace via the service client itself is the only way I found to get it.